spyder/app/mainwindow.py

```diff
@@ -1273,6 +1273,18 @@ def add_ipm_action(text, path):
                 except TypeError:
                     pass
 
+        # Right justify keyboard shortcuts.
+        # Apply only for none dark themes, otherwise tests fail.
+        if not is_dark_interface():

```

Comment:
This means that your fix won't apply to the dark theme. Since that's going to be the default theme in Spyder 4, that sort of defeats the purpose of your work (which I find very nice).

So instead of using `is_dark_interface` in this PR, please use instead `running_under_pytest` (as I instructed you to do it before).